src/consumer.rs

```diff
@@ -309,9 +316,10 @@ pub async fn from_substrait_rel(
                     }
                 })
                 .collect::<Result<Vec<_>>>()?;
-            let left_cols: Vec<&str> = pairs.iter().map(|(l, _)| l.as_str()).collect();
-            let right_cols: Vec<&str> = pairs.iter().map(|(_, r)| r.as_str()).collect();
-            left.join(right, join_type, &left_cols, &right_cols, None)
+            let left_cols: Vec<Column> = pairs.iter().map(|(l, _)| l.clone()).collect();
+            let right_cols: Vec<Column> = pairs.iter().map(|(_, r)| r.clone()).collect();
```

Comment:
Would this also work?

```Rust
let (left_cols, right_cols): (Vec<_>, Vec<_>) = pairs.iter().cloned().unzip();
```

Reply:
Thanks. Yes, that is much nicer. Applied.